### PR TITLE
ApiProvider: correctly register listeners, allow to disable listeners

### PR DESCRIPTION
**Pull request by phryneas**
As mentioned in #2003, might even be worth to put it into a 1.8 patch depending how long we want to wait until 1.9.

### CODE REVIEW

File: packages/toolkit/src/query/react/ApiProvider.tsx [hunk 1]
```diff
@@ -45,7 +45,13 @@ export function ApiProvider<A extends Api<any, {}, any, any>>(props: {
...
   )
   // Adds the event listeners for online/offline/focus/etc
-  setupListeners(store.dispatch, props.setupListeners)
+  useEffect(
+    (): undefined | (() => void) =>
+      props.setupListeners === false
+        ? undefined
+        : setupListeners(store.dispatch, props.setupListeners),
+    [props.setupListeners]
+  )
```
Comment (msutkowski) [inline packages/toolkit/src/query/react/ApiProvider.tsx:54]: Hmmm, do we need to rework `setupListeners` and `initialized` make sure we're cleaning up the subscriptions here if we're putting it in an effect?

Comment (markerikson) [reply packages/toolkit/src/query/react/ApiProvider.tsx:54]: Actually, that's a good point. Especially with React 18 double-running effects in dev.

Comment (phryneas) [reply packages/toolkit/src/query/react/ApiProvider.tsx:54]: `setupListeners` returns an `unsubscribe` function so this should be fine as it is.

**Comment by markerikson** [comment]
Let's put this in `master` now.

**Pull request merged**